controllers/openstackdataplanenodeset_controller.go

```diff
@@ -193,6 +199,16 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 
 	instance.Status.Conditions.MarkFalse(dataplanev1.SetupReadyCondition, condition.RequestedReason, condition.SeverityInfo, condition.ReadyInitMessage)
 
+	// Detect config changes and set Status ConfigHash
+	configHash, err := r.GetSpecConfigHash(instance)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if configHash != instance.Status.DeployedConfigHash {
```

Comment:
This is kind of redundant, though there could be reconciliation when services are updated in a `OpenStackDataPlaneNodeSet`. But I don't know why someone would do that when they can override it in `OpenStackDataPlaneDeployment`. That's the reason I had proposed to move services to `OpenStackDataPlaneDeployment`.

Also, the hash should include `baremetalSetTemplate` if exists.

Reply:
Also, I noticed we're adding (extra) ansible vars to `OpenStackDataPlaneDeployment` with https://issues.redhat.com/browse/OSPRH-1141, so how are these hashes relevant in situations where users use it?

Reply:
`baremetalSetTemplate` will be reconciled regardless of the `OpenStackDataPlaneDeployment` though won't it? Like, we don't need a `OpenStackDataPlaneDeployment` to exist for it to provision the nodes.

The issue we ran into that led to this PR was that users might change some Ansible variable in the NodeSet and expect that it is reconciled. But in reality, they actually need to create a new `OpenStackDataPlaneDeployment` for those changes to be applied. So this `configHash` and `deployedConfigHash` are designed to give the user some visual indication that there is config drift, and the new config has not been applied.

I think @slagle has some use cases around updates and upgrades as well that he is considering.

Reply:
> baremetalSetTemplate will be reconciled regardless of the OpenStackDataPlaneDeployment though won't it?

If we're looking for `OpenStackDataPlaneNodeSet` changes to run a new `OpenStackDataPlaneDeployment`, I think it could be anything including services in `OpenStackDataPlaneNodeSet` and update-able fields fields in  `baremetalSetTemplate`, no?

Reply:
The point of this PR was to record the hashes as a visual indication if something had changed since the last Deployment, not to automatically start a new Deployment. I'd leave that as a future enhancement where we can discuss how that might work.

As for services and vars on the Deployment itself, I would only record Status.DeployConfigHash if the completed Deployment does not have services or vars set.

Reply:
I'd probably add that logic to `checkDeployment`. `isDeploymentReady` and `deploymentExists` should probably only return True if the Deployment was for the services and vars on the NodeSet.